--- a/source/input.c
+++ b/source/input.c
@@ -500,6 +500,20 @@ int input_read_parameters(
       pba->H0 = param1;
     }
 
+    /* TEP parameters */
+    class_call(parser_read_string(pfc, "tep_mode", &string1, &flag1, errmsg),
+               errmsg, errmsg);
+    if (flag1 == _TRUE_) {
+      if (strcmp(string1, "yes") == 0) {
+        pba->tep_mode = _TRUE_;
+      }
+    }
+
+    class_call(parser_read_double(pfc, "Sigma_0", &param1, &flag1, errmsg),
+               errmsg, errmsg);
+    if (flag1 == _TRUE_) {
+      pba->Sigma_0 = param1;
+    }
+
     /* ... */
   }
   /* ... */
